dashboard/app/controllers/api/v1/sections_controller.rb

```diff
@@ -131,4 +171,11 @@ def find_follower
     end
     @follower = Follower.where(section: @section.id, student_user_id: current_user.id).first
   end
+
+  # Update script_id if user provided valid script_id
+  # Set script_id to nil if invalid or no script_id provided
+  def set_script_id(script_id)
+    return script_id if Script.valid_script_id?(current_user, script_id)
+    nil
+  end
```

Comment:
The original endpoint in pegasus (`PATCH /v2/sections/:id`) set the `script_id` for a section to nil if not explicitly supplied by the consumer, so this method preserves that functionality. If a user supplies a valid `script_id`, it will be set on the section, but if they supply an invalid or no `script_id`, it is set to nil.

There is a unit test that covers this case [here](https://github.com/code-dot-org/code-dot-org/blob/0429898d70186f27201f596cb1ca407a1eedb0ef/dashboard/test/controllers/api/v1/sections_controller_test.rb#L557), but it was previously not catching the bug because `Script.valid_script_id?` was stubbed to always return true. I moved this assertion only to the places that need it, which caught the bug 👍